### PR TITLE
JAX: Add Seq2Seq

### PR DESCRIPTION
**Pull request by AnirudhDagar**
*Description of changes:*

By submitting this pull request, I confirm that you can use, modify,
copy, and redistribute this contribution, under the terms of your
choice.

### CODE REVIEW

File: chapter_recurrent-modern/seq2seq.md [hunk 1]
```diff
@@ -453,11 +527,35 @@ where the last dimension of the tensor stores the predicted token distribution.
...
+if tab.selected('mxnet', 'pytorch', 'tensorflow'):
+    state = decoder.init_state(encoder(X))
+    outputs, state = decoder(X, state)
+if tab.selected('jax'):
+    state = decoder.init_state(encoder.init_with_output(d2l.get_key(), X)[0])
+    (outputs, state), _ = decoder.init_with_output(d2l.get_key(), X, state)
+
+
+d2l.check_shape(outputs, (batch_size, num_steps, vocab_size))
+if tab.selected('mxnet', 'pytorch', 'jax'):
```
Comment (astonzhang) [inline chapter_recurrent-modern/seq2seq.md:558]: redundant?

File: chapter_recurrent-modern/seq2seq.md [hunk 2]
```diff
@@ -522,32 +637,75 @@ of any irrelevant prediction
...
     decoder = Seq2SeqDecoder(
         len(data.tgt_vocab), embed_size, num_hiddens, num_layers, dropout)
+if tab.selected('mxnet', 'pytorch'):
     model = Seq2Seq(encoder, decoder, tgt_pad=data.tgt_vocab['<pad>'],
                     lr=0.001)
-    trainer = d2l.Trainer(max_epochs=50, gradient_clip_val=1, num_gpus=1)
+if tab.selected('jax'):
+    model = Seq2Seq(encoder, decoder, tgt_pad=data.tgt_vocab['<pad>'],
+                    lr=0.001, training=True)
+    trainer = d2l.Trainer(max_epochs=70, gradient_clip_val=1, num_gpus=1)
```
Comment (astonzhang) [inline chapter_recurrent-modern/seq2seq.md:708]: redundant?

File: chapter_recurrent-modern/seq2seq.md [hunk 3]
```diff
@@ -522,32 +637,75 @@ of any irrelevant prediction
...
+
 ## [**Training**]
 :label:`sec_seq2seq_training`
 
 Now we can [**create and train an RNN encoder-decoder model**]
 for sequence to sequence learning on the machine translation dataset.
 
 ```{.python .input}
-%%tab all
+%%tab pytorch, tensorflow, mxnet
```
Comment (astonzhang) [inline chapter_recurrent-modern/seq2seq.md:670]: tab all? tab.selected('jax') is used below

**Review by astonzhang — CHANGES_REQUESTED** [review]

File: chapter_recurrent-modern/seq2seq.md (hunk 1 above)
Comment (AnirudhDagar) [reply chapter_recurrent-modern/seq2seq.md:558]: What is redundant?

Comment (AnirudhDagar) [reply chapter_recurrent-modern/seq2seq.md:558]: Ohh, i see, somehow a couple of tabs were copied. Turns out the final commit doesn't include the final state of the saved notebook on my local machine. Sorry my bad.

**Pull request merged**